Src/PerformanceCollector/Shared/Implementation/QuickPulse/QuickPulseEventSource.cs

```diff
@@ -0,0 +1,134 @@
+﻿namespace Microsoft.ApplicationInsights.Extensibility.PerfCounterCollector.Implementation.QuickPulse
+{
+    using System;
+#if !NET40
+    using System.Diagnostics.Tracing;
+#endif
+
+#if NET40
+    using Microsoft.Diagnostics.Tracing;
+#endif
+
+    [EventSource(Name = "Microsoft-ApplicationInsights-Extensibility-PerformanceCollector-QuickPulse")]
+    internal sealed class QuickPulseEventSource : EventSource
+    {
+        private static readonly QuickPulseEventSource Logger = new QuickPulseEventSource();
+
+        private QuickPulseEventSource()
+        {
+            this.ApplicationName = this.GetApplicationName();
+        }
+
+        public static QuickPulseEventSource Log
+        {
+            get
+            {
+                return Logger;
+            }
+        }
+
+        public string ApplicationName { [NonEvent]get; [NonEvent]private set; }
+
+        #region Infra init - success
+
+        [Event(1, Level = EventLevel.Informational, Message = @"QuickPulse infrastructure is being initialized. {0}")]
+        public void ModuleIsBeingInitializedEvent(
+            string message,
+            string applicationName = "dummy")
+        {
+            this.WriteEvent(1, message, this.ApplicationName);
```

Comment:
I'm not sure abot all versions but in some EventSource does not properly handles null values. It would be great if we add ?? string.Empty to all places where you use strings